File: Python/Invert Binary Tree.py
class Solution:
    # @param {TreeNode} root
    # @param {TreeNode}
    def invertTree(self, root):
        if root != None:
            tempPtr = root.left
            root.left = root.right
            root.right = tempPtr
            self.invertTree(root.right)
            self.invertTree(root.left)
        return root

File: java/Invert_Binary_Tree.java
// Google: 90% of our engineers use the software you wrote (Homebrew), 
// but you can’t invert a binary tree on a whiteboard so fuck off.
// Recursive solution.
// Time is O(n), space is O(n).

public class Solution {
    public TreeNode invertTree(TreeNode root) {
        invert(root);
        return root;
    }
    public void invert(TreeNode root) {
        if (root == null) return;
        TreeNode left = root.left, right = root.right;
        root.left = right;
        root.right = left;
        invert(left);
        invert(right);
    }
}
